### PR TITLE
Fix compiler errors and warnings with C-XSC support

### PR DESCRIPTION
**Pull request by jamesjer**
I attempted to build version 1.0.1 for Fedora.  We build with C-XSC support.  The build failed like this:
```
cxsc_poly.C:18:15: error: conflicting declaration of 'int cpoly_CXSC(int, const cxsc::complex*, cxsc::complex*, int)' with 'C' linkage
   18 | #define cpoly cpoly_CXSC
      |               ^~~~~~~~~~
cpoly.C:339:16: note: in expansion of macro 'cpoly'
  339 | extern "C" int cpoly(int degree, const xcomplex *poly, xcomplex *Roots, int prec)
      |                ^~~~~
In file included from cxsc_poly.C:16:
cxsc_poly.h:1:5: note: previous declaration with 'C++' linkage
    1 | int cpoly_CXSC(int degree, const cxsc::complex coeffs[], cxsc::complex roots[], int prec);
      |     ^~~~~~~~~~
```
The first commit in this series fixes that by removing the `extern "C"`.  I was able to build with support for all of mpfr, mpfi, mpc, fplll, and cxsc with that change.

The second commit fixes this warning:
```
floattypes.h:165:5: warning: 'cpoly_CXSC' violates the C++ One Definition Rule [-Wodr]
  165 | int cpoly_CXSC(int degree, cxsc::complex coeffs[], cxsc::complex roots[], int prec);
      |     ^
cpoly.C:339:16: note: type mismatch in parameter 2
  339 | extern "C" int cpoly(int degree, const xcomplex *poly, xcomplex *Roots, int prec)
      |                ^
cpoly.C:339:16: note: type 'const struct xcomplex' itself violates the C++ One Definition Rule
/usr/include/cxsc/complex.hpp:49:7: note: the incompatible type is defined here
   49 | class complex
      |       ^
cpoly.C:339:16: note: 'cpoly_CXSC' was previously declared here
```

The third commit fixes the warning about parameter 2 of cpoly noted above.

The fourth commit fixes this warning:
```
mpc.c:653:7: warning: type of 'cpoly_MPC' does not match original declaration [-Wlto-type-mismatch]
  653 |   int cpoly_MPC(int, mpc_t *, mpc_t *, mp_prec_t);
      |       ^
cpoly.C:339:16: note: type mismatch in parameter 4
  339 | extern "C" int cpoly(int degree, const xcomplex *poly, xcomplex *Roots, int prec)
      |                ^
cpoly.C:339:16: note: type 'int' should match type 'mpfr_prec_t'
cpoly.C:339:16: note: 'cpoly_MPC' was previously declared here
cpoly.C:339:16: note: code may be misoptimized unless '-fno-strict-aliasing' is used
```

On various platforms, `mp_prec_t` may be a typedef of int, long, short, or unsigned short.

### CODE REVIEW

File: src/mpc.c
Comment (fingolfin) [inline src/mpc.c:653]: In addition I would suggest to avoid function declarations like this inside a function. Instead, I'd move it to a new file `mpc_poly.h` (similar to `cxsc_poly.h`) which then can be included by both `mpc.c` and `mp_poly.C`. 

Of course this is not a criticism of this PR nor does it need to be addressed in this PR; I merely was "inspired" by reviewing this code.

**Review by fingolfin — APPROVED** [review]
Thank you! Very clean and clear. I recommend merging it, but will leave this to the maintainer, @laurentbartholdi

**Comment by laurentbartholdi** [comment]
I cleaned up a little the whole code. It's still ugly (and non-reentrant and non-parallelizable) though.

**Pull request merged**
